applicationset/services/scm_provider/gitlab.go

```diff
@@ -195,3 +195,46 @@ func (g *GitlabProvider) listBranches(_ context.Context, repo *Repository) ([]gi
 	}
 	return branches, nil
 }
+
+func GitlabListTags(_ context.Context, repo *Repository, g *GitlabProvider) ([]gitlab.Tag, error) {
+	tags := []gitlab.Tag{}
+	sortType := "desc"
+	orderBy := "version"
+	opt := &gitlab.ListTagsOptions{
+		ListOptions: gitlab.ListOptions{PerPage: 100},
+		Sort:        &sortType,
+		OrderBy:     &orderBy,
+	}
+	for {
+		gitlabTags, resp, err := g.client.Tags.ListTags(repo.RepositoryId, opt)
+		// 404s are not an error here, just a normal false.
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			return []gitlab.Tag{}, nil
+		}
```

Comment:
Do we get an `err` when receiving 404s? If not, can we check for `err != nil` first?